LSPS0/README.md

```diff
@@ -0,0 +1,579 @@
+> compression greatly simplifies implementation, testing,
+> interoperability, and dependencies.
+
+The client:
+
+* MUST send single complete JSON-RPC 2.0 request objects,
+  UTF-8-encoded, as payloads for message ID 37913.
+* MUST NOT send JSON-RPC 2.0 notification objects
+  (i.e. every object it sends must have an `id` field).
+* MUST NOT use by-position parameter structures, and MUST
+  use by-name parameter structures.
+* MUST NOT batch multiple JSON-RPC 2.0 request objects in
+  an array as described in the "Batch" section of the
+  JSON-RPC 2.0 specification.
+
+> **Rationale** By disallowing by-position parameter
+> structures, other LSPS specifications need only to define
+> parameter names and not some canonical order of parameters
+> for by-position use.
+> Having to handle only by-name parameter structures also
+> simplifies the LSP code, as it does not have to check
+> whether the `params` value is an array or a dictionary,
+> and to separately map array elements to `params` keys.
+> Batched requests require batched responses according to
+> JSON-RPC 2.0, and it may be simpler for LSPs to handle
+> unrelated LSPS methods separately without requiring
+> re-batching of the responses; it gives a simple "one
+> message is one request / response" rule.
+
+The LSP:
+
+* MUST send single complete JSON-RPC 2.0 response objects,
+  UTF-8-encoded, as payloads for message ID 37913.
+* MAY send JSON-RPC 2.0 notification objects
+  (i.e. it may send an object without `id` but with `method`,
+  `params`, and `jsonrpc` fields).
+  * MUST use by-name parameter structures for notifications.
+* MUST respond with an error -32602 if a request `params` field
+  is not recognized for the given `method`.
+* SHOULD NOT send message ID 37913 unless the peer already sent a
+  message ID 37913, possibly in a past connection session.
+
+> **Rationale** The peer sending message ID 37913 is an
+> indicator that it understands the LSPS0 protocol and wishes
+> to act as a client.
+> If the peer does not send it, then it is not an LSPS client
+> and the LSP has no reason to send message ID 37913.
+> Notifications allow the LSP to initiate with the client,
+> such as if the client has an incoming payment that the LSP
+> learns about.
+
+JSON-RPC 2.0 is an asynchronous protocol, and the order
+in which the LSP sends responses MAY be different from the
+order in which the client sent the requests.
+
+Other LSPS specifications:
+
+* MUST indicate `method` names for each API endpoint and each
+  LSP-initiated notification, as well as the key names of the
+  `params` for each `method`, and the meanings of each
+  parameter.
+  * MUST prefix `method` names:
+    * For `method`s for client requests, prefix with `lsps`
+      followed by the LSPS number followed by `.`, e.g.
+      `lsps999.dothis`.
+    * For `method`s for LSP notifications, prefix with
+      `n.lsps` followed by the LSPS number followed by `.`,
+      e.g. `n.lsps999.thathappened`.
```

Comment:
What is the purpose of `n.`?

Reply:
To separate the namespace for notifications from the namespace for actual requests and responses.

Reply:
In meeting, I believe Sev accepted this.